### PR TITLE
Make laser_geometry build for ros2 (on windows 10)

### PR DESCRIPTION
**Pull request by bfjelds**
Should go to a ROS2 branch, but I'm not sure how to create a PR for that :)

Working towards enabling laser_filters in ROS2. This provides ability to build and updates for ROS2 constructs (like Node vs NodeHandle).

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,47 +1,51 @@
...
 include_directories(include
-    ${catkin_INCLUDE_DIRS}
-    ${Boost_INCLUDE_DIRS}
-    ${Eigen_INCLUDE_DIRS}
+    ${angles_INCLUDE_DIRS}
+    ${rclcpp_INCLUDE_DIRS}
+    ${sensor_msgs_INCLUDE_DIRS}
+    ${tf2_INCLUDE_DIRS}
+    ${tf2_eigen_INCLUDE_DIRS}
+    $ENV{BOOST_ROOT}
```
Comment (bfjelds) [inline CMakeLists.txt]: use find_package to find BOOST

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,47 +1,51 @@
...
 
-find_package(Boost REQUIRED COMPONENTS system thread)
+find_package(ament_cmake REQUIRED)
 
-find_package(Eigen REQUIRED)
+find_package(angles REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(sensor_msgs REQUIRED)
+find_package(tf2 REQUIRED)
+find_package(ecl_eigen REQUIRED)
```
Comment (bfjelds) [inline CMakeLists.txt]: use Eigen instead of ecl_eigen?

**Comment by jonbinney** [comment]
+1 for this. I'd like to wait until this is merged before merging https://github.com/ros-perception/laser_filters/pull/61

**Comment by jonbinney** [comment]
@wjwwood if you're OK with it, i can go ahead and make a ros2 branch for this repo, and start merging PRs into it. That will let me merge dependent PRs in the laser_filters package.

**Comment by wjwwood** [comment]
@jonbinney yeah, go for it. Sorry for the late reply, and thanks for helping out.

**Comment by greimela-si** [comment]
One thing I noticed is that there are still boost dependencies used in the code.
The official [migration guide](https://github.com/ros2/ros2/wiki/Migration-Guide) suggests to remove boost where possible.

**Comment by Martin-Idel-SI** [comment]
There are a couple of things I noticed while trying to build this on Linux (I want to migrate the LaserScanDisplay for rviz). Maybe this can help you:
- C++14 needs to be set. This can be achieved by adding three lines to CMakeLists:
```
if(NOT CMAKE_CXX_STANDARD)
  set(CMAKE_CXX_STANDARD 14)
endif()
```
- The capitalization of includes for sensor_msgs is wrong: The correct include would be e.g. `#include "sensor_msgs/msg/laser_scan.hpp"` instead of `#include "sensor_msgs/msg/Laser_Scan.hpp"` (also, I'd choose either `<>` type includes or `""` type includes but not both).
- The tests require the `angles` package. If you only have ros2 on your system, this doesn't compile, yet, as it's only available as catkin package.

**Comment by bfjelds** [comment]
Here is a PR for ROS2 angles: https://github.com/ros/angles/pull/12

**Comment by jonbinney** [comment]
@bfjelds I've created a ros2 branch for this repo; can you re-target the PR?

**Comment by jonbinney** [comment]
@bfjelds nevermind, william showed me that I can change the target branch.

**Pull request merged**